src/typing_extensions.py

```diff
@@ -829,35 +853,73 @@ def __new__(cls, name, bases, ns, total=True):
                 }
             required_keys = set()
             optional_keys = set()
+            readonly_keys = set()
+            mutable_keys = set()
 
             for base in bases:
                 annotations.update(base.__dict__.get('__annotations__', {}))
                 required_keys.update(base.__dict__.get('__required_keys__', ()))
                 optional_keys.update(base.__dict__.get('__optional_keys__', ()))
+                readonly_keys.update(base.__dict__.get('__readonly_keys__', ()))
+                mutable_keys.update(base.__dict__.get('__mutable_keys__', ()))
+
+                if not getattr(base, "__other_keys__", True):
+                    if (
+                        set(own_annotations) -
+                        set(base.__dict__.get('__annotations__', {}))
```

Comment:
runs about twice as fast:

```suggestion
                        own_annotations.keys() -
                        base.__dict__.get('__annotations__', {}).keys()
```

Reply:
Removed this code